arviz/wrappers/wrap_pystan.py

```diff
@@ -55,3 +55,85 @@ def get_inference_data(self, fit):
     def log_likelihood__i(self, excluded_obs_log_like, idata__i):
         """Retrieve the log likelihood of the excluded observations from ``idata__i``."""
         return idata__i.log_likelihood[excluded_obs_log_like]
+
+
+class PyStan3SamplingWrapper(SamplingWrapper):
```

Comment:
I was just thinking that we could have an intermediate layer `StanSamplingWrapper` to avoid having to repeat the `sel_observations` and `log_likelihood__i` in pystan, pystan3 and cmdstanpy sampling wrappers, as the base logic of generating log likelihoods during sampling is the same for all 3. We can do that in a follow up PR or directly when we implement the cmdstanpy base wrapper.